Code/Framework/AzCore/AzCore/DOM/DomPrefixTree.inl

```diff
@@ -328,9 +385,33 @@ namespace AZ::Dom
         }
     }
 
+    template<class T>
+    DomPrefixTree<T> DomPrefixTree<T>::DetachSubTree(const Path& path)
+    {
+        Node node = DetachNodeAtPath(path);
+        if (!node.IsEmpty())
+        {
+            return DomPrefixTree<T>(AZStd::move(node));
+        }
+        return DomPrefixTree<T>();
```

Comment:
nit: seems like this can be a one-liner `return DomPrefixTree<T>(AZStd::move(node));` since `DetachNodeAtPath` always returns a valid node.

Reply:
I'll update this in the next PR since this one is ready for a merge